### PR TITLE
Migrated server code to TypeScript

### PR DESCRIPTION
**Pull request by Danbaba1**
* **Converted server-side JavaScript code to TypeScript for improved type safety and maintainability.**
* **Updated server to use TypeScript for enhanced static typing and potential code improvements.**

### CODE REVIEW

**Comment by Danbaba1** [comment]
**Please review my pull request** @collabo-community/review-backend-api

File: server/src/server.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { app as app} from './app'
+import http from 'http';
+// Interface for API details
+
+
+const PORT = parseInt(process.env.PORT || '3000', 10); // Ensure PORT is a number
```
Comment (Ifycode) [inline server/src/server.ts:6]: Nice work so far @Danbaba1 

But there is no need to check PORT here. Just do it as you see it is done in our templates.

File: server/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,108 @@
+{
+  "compilerOptions": {
```
Comment (Ifycode) [inline server/tsconfig.json:2]: For tsconfig, is this the exact configuration in our template?
If not, you want to remove what's here and copy the contents of our template's tsconfig here instead @Danbaba1

File: server/package.json [hunk 3]
```diff
@@ -1,18 +1,24 @@
 {
-  "type": "module",
   "dependencies": {
```
Comment (Ifycode) [inline server/package.json:2]: I see dependencies coming first and the order of things generally rearranged in your package.json here. We want the order of things in the package.json to be exactly the same as that of our templates. Make the changes accordingly @Danbaba1

File: server/package.json [hunk 4]
```diff
@@ -1,18 +1,24 @@
...
   "keywords": [],
   "author": "",
-  "license": "ISC"
+  "license": "ISC",
+  "devDependencies": {
+    "@types/express": "^4.17.21",
+    "@types/node": "^20.14.8",
+    "nodemon": "^3.1.4",
+    "ts-node": "^10.9.2",
+    "typescript": "^5.5.2"
```
Comment (Ifycode) [inline server/package.json:22]: For both dependencies and devDependecies in your package.json. As long as you can find it in our own template, use the same exact versions you find there in here @Danbaba1

File: server/package.json [hunk 5]
```diff
@@ -1,18 +1,24 @@
 {
-  "type": "module",
   "dependencies": {
     "dotenv": "^16.4.5",
     "express": "^4.19.2",
     "node-fetch": "^3.3.2"
   },
   "scripts": {
-    "start": "node server.js"
+    "start": "nodemon --exec ts-node src/server.ts"
```
Comment (Ifycode) [inline server/package.json:8]: @Danbaba1 Eslint: 
- `lint` and `lint-watch` scripts are missing in your package.json (and in your dependencies)
- Also all files that configures or that makes eslint work are not present

> Ensure that eslint alerts coding issues in this server also, just like it does in our template.

File: server/src/app.ts [hunk 6]
```diff
@@ -0,0 +1,14 @@
+import express, { Express, Request, Response } from 'express';
+import { router as appRouter } from './api/routes/app.route';
+import dotenv from 'dotenv';
+
+dotenv.config();
+
+const app: Express = express();
+
```
Comment (Ifycode) [inline server/src/app.ts:8]: Morgan is missing here and as a package. We want Morgan to give status report for our API in the terminal. See our template for how Morgan works and is implemented @Danbaba1

File: server/src/app.ts [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+app.use(express.urlencoded({ extended: false }));
+app.use(express.json());
+
```
Comment (Ifycode) [inline server/src/app.ts:11]: Wondering why you left out the CORS part here and in the package.json @Danbaba1 
Implement it as done in our template.

**Review by Ifycode — CHANGES_REQUESTED** [review]

File: server/src/server.ts (hunk 1 above)
Comment (Danbaba1) [reply server/src/server.ts:6]: I will make the necessary changes ma.

File: server/tsconfig.json (hunk 2 above)
Comment (Danbaba1) [reply server/tsconfig.json:2]: I will make the necessary changes ma.

File: server/package.json (hunk 3 above)
Comment (Danbaba1) [reply server/package.json:2]: I will make the necessary changes ma.

File: server/package.json (hunk 4 above)
Comment (Danbaba1) [reply server/package.json:22]: I will make the necessary changes ma.

File: server/package.json (hunk 5 above)
Comment (Danbaba1) [reply server/package.json:8]: I will make the necessary changes ma.

File: server/src/app.ts (hunk 6 above)
Comment (Danbaba1) [reply server/src/app.ts:8]: I will make the necessary changes ma.

File: server/src/app.ts (hunk 7 above)
Comment (Danbaba1) [reply server/src/app.ts:11]: I will make the necessary changes ma.

**Comment by Danbaba1** [comment]
Made necessary changes as requested. Please review PR. @collabo-community/review-backend-api

**Comment by Danbaba1** [comment]
https://discord.com/channels/1148391778492354741/1253584233155395657/1253633682821091438

**Review by Ifycode — CHANGES_REQUESTED** [review]
You also left out some files like the `editorconfig`, `.env.example` etc.
@Danbaba1 please thoroughly check that you have added all the necessary files before you request for review again. You should be making this server as close as possible to our template, not waiting for me to tell you to add them.

For the `.env.example` file just modify it slightly (logically) based on the environment variables we are using in this case.

File: package-lock.json [hunk 8]
```diff
@@ -12,8 +12,7 @@
         "dotenv": "^16.4.5",
         "express": "^4.19.2",
         "node-fetch": "^3.3.2"
-      },
-      "devDependencies": {}
```
[the remark this answers is not in the document]
Comment (Danbaba1) [reply package-lock.json:16]: Okay ma. Noted

**Comment by Danbaba1** [comment]
Don't need it again.

**Pull request closed without merging**